datadog/resource_datadog_dashboard.go

```diff
@@ -4722,6 +4722,257 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
+										},
+									},
+								},
+							},
+							"group_by": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Description: "Group by options.",
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"facet": {
+											Type:        schema.TypeString,
+											Required:    true,
+											Description: "Event facet.",
+										},
+										"limit": {
+											Type:        schema.TypeInt,
+											Optional:    true,
+											Description: "Number of groups to return.",
+										},
+										"sort": {
+											Type:        schema.TypeList,
+											Optional:    true,
+											MaxItems:    1,
+											Description: "Options for sorting group by results.",
+											Elem: &schema.Resource{
+												Schema: map[string]*schema.Schema{
+													"aggregation": {
+														Type:        schema.TypeString,
+														Required:    true,
+														Description: "Aggregation methods for event platform queries.",
+													},
+													"metric": {
+														Type:        schema.TypeString,
+														Optional:    true,
+														Description: "Metric used for sorting group by results.",
+													},
+													"order": {
+														Type:        schema.TypeString,
+														Optional:    true,
+														Description: "Direction of sort.",
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+							"name": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Name of query for use in formulas.",
+							},
+						},
+					},
+				},
+				"process_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "Process query using formulas and functions.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Data source for process queries.",
+							},
+							"metric": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Process metric name.",
+							},
+							"text_filter": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Description: "Text to use as filter.",
+							},
+							"tag_filters": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+								Description: "An array of tags to filter by.",
+							},
+							"limit": {
+								Type:        schema.TypeInt,
+								Optional:    true,
+								Description: "Number of hits to return.",
+							},
+							"sort": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Description: "Direction of sort.",
+								Default:     "desc",
+							},
+							"aggregator": {
```

Comment:
Missing enum validation.

Reply:
Done in [commit 2f914c9](https://github.com/DataDog/terraform-provider-datadog/pull/892/commits/2f914c9da8e833c9a46c5c24bd654aa47ba44918)